_posts/2020-07-15-#19031.md

```diff
@@ -0,0 +1,51 @@
+---
+layout: pr
+date: 2020-07-15
+title: "Implement ADDRv2 support (part of BIP155)"
+pr: 19031
+authors: [vasild]
+components: ["p2p"]
+host: troygiorshev
+status: upcoming
+commit:
+---
+
+## Notes
+
+* [PR 19031](https://github.com/bitcoin/bitcoin/pull/19031) is the implementation of [BIP 155](https://github.com/bitcoin/bips/blob/9286b5254317d9e73fb25c5f0acd2b2d9937843e/bip-0155.mediawiki), the `addrv2` message.  PR 19031 is being used as a roadmap, where [PR 19360](https://github.com/bitcoin/bitcoin/pull/19360) is currently ready for review.
+
+* The `addr` message is currently used to relay IP addresses across the network.  It uses only IPv6 addresses, where IPv4 addresses are given as IPv4-mapped IPv6 addresses.  More information about it can be found [here](https://btcinformation.org/en/developer-reference#addr).  The `addrv2` message will allow for longer variable length addresses.
+
+* Bitcoin currently has Tor integration.  Relevant to this PR is the fact that Tor v2 addresses are 16 characters long, nicely fitting in the 16 bytes needed for an IPv6 address.  Ultimately these changes will allow us to work with [Tor v3](https://github.com/bitcoin/bitcoin/issues/18884), along with many other different types of networks.
+
+* PR 19031 is large and is still a WIP.  There's no reason to review all of it in detail.  Here is a suggested approach:
+  * Review [PR 19360](https://github.com/bitcoin/bitcoin/pull/19360) (this is also the first commit of PR 19031)
+  * Read [BIP 155](https://github.com/bitcoin/bips/blob/9286b5254317d9e73fb25c5f0acd2b2d9937843e/bip-0155.mediawiki).  There is ongoing discussion on [BIP 155](https://github.com/bitcoin/bips/blob/9286b5254317d9e73fb25c5f0acd2b2d9937843e/bip-0155.mediawiki) happening at [BIP PR 907](https://github.com/bitcoin/bips/pull/907).  Because of this, please review [BIP 155](https://github.com/bitcoin/bips/blob/9286b5254317d9e73fb25c5f0acd2b2d9937843e/bip-0155.mediawiki) at any of the links in these notes as they reflect the current status of the BIP.
+  * See [this comment](https://github.com/bitcoin/bips/pull/907#issuecomment-611924492) in BIP PR 907 about how to signal support for `addrv2`.  The final decision can be seen [here](https://github.com/bitcoin/bips/blob/9286b5254317d9e73fb25c5f0acd2b2d9937843e/bip-0155.mediawiki#Signaling_support_and_compatibility)
+  * Read through all of the conceptual comments in 19031
+  * Skim commits 2-4 of PR 19031.  The goal should be to get context for what future PRs will cover, and to see how the implementation of [BIP 155](https://github.com/bitcoin/bips/blob/9286b5254317d9e73fb25c5f0acd2b2d9937843e/bip-0155.mediawiki) matches the specification.
```

Comment:
For each of the items in this list, maybe use a full stop consistently or not at all.